lib/espree.js

```diff
@@ -98,6 +100,8 @@ export default () => Parser => {
                 }
             }, code);
 
+            this.originalSourceType = originalSourceType || options.sourceType;
```

Comment:
It may be safer to use a symbol key, or store `originalSourceType` on `this[STATE]` instead of directly on `this`.

Reply:
Can you explain what you mean by safer?

Reply:
Okay, I'm assuming you mean that it's safer just in case it collides with an Acorn property? In that case, I agree, I've moved into `this[STATE]`.